runtime/pyproject.toml

```diff
@@ -7,6 +7,5 @@ requires = [
     "packaging",
     "pybind11>=2.10.1",
     "PyYAML",
-    "requests",
```

Comment:
Something in here is still using this:

https://github.com/openxla/iree/actions/runs/4370994923/jobs/7646455762#step:7:250
```
 82/110 Test #115: iree/build_tools/benchmarks/post_benchmark_comment_test ................................***Failed    0.18 sec
Traceback (most recent call last):
  File "C:\a\iree\iree\build_tools\benchmarks\post_benchmark_comment_test.py", line 9, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
```

Reply:
Adding back in.